native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.android.js

```diff
@@ -0,0 +1,241 @@
+    iconStyle: {
+      color: colorRed500,
+    },
+  },
+  [ALERT_TYPES.NEUTRAL]: {
+    iconSource: 'information-circle',
+    iconStyle: {
+      color: colorGray500,
+    },
+  },
+};
+
+const BpkBannerAlert = props => {
+  const {
+    type,
+    message,
+    onDismiss,
+    onToggleExpanded,
+    dismissable,
+    expanded,
+    dismissButtonLabel,
+    toggleExpandedButtonLabel,
+    show,
+    style: userStyle,
+    animateOnEnter,
+    animateOnLeave,
+    children,
+    ...rest
+  } = props;
+
+  const expandable = children !== null;
+  const { iconSource, iconStyle } = ALERT_TYPE_STYLES[type] || {};
+
+  const dismissableContent = (
+    <TouchableNativeFeedback
+      accessibilityComponentType="button"
+      accessibilityLabel="dismiss"
+      background={TouchableNativeFeedback.SelectableBackgroundBorderless()}
+      useForeground
+      onPress={onDismiss}
+    >
+      <View style={styles.actionContainer}>
+        <BpkText textStyle="sm" emphasize style={styles.dismiss}>
+          DISMISS
+        </BpkText>
+      </View>
+    </TouchableNativeFeedback>
+  );
+  const expandableContent = (
+    <BpkAnimateHeight expanded={expanded}>
+      <View style={styles.childrenContainer}>{children}</View>
+    </BpkAnimateHeight>
+  );
+  const expandableIcon = (
+    <View style={styles.dismissableContainer}>
+      <BpkIcon
+        style={styles.icon}
+        icon={expanded ? 'chevron-up' : 'chevron-down'}
+        small
+      />
+    </View>
+  );
+  const mainContent = (
+    <View style={styles.container}>
+      <View style={styles.mainContainer}>
+        <View style={styles.container}>
+          <BpkIcon style={[styles.icon, iconStyle]} icon={iconSource} small />
+          <BpkText style={styles.text} textStyle="sm">
+            {message}
+          </BpkText>
+        </View>
+        {expandable && expandableContent}
+      </View>
+      {dismissable && dismissableContent}
+      {expandable && expandableIcon}
+    </View>
+  );
+  return (
+    <AnimateAndFade
+      animateOnEnter={animateOnEnter}
+      animateOnLeave={dismissable || animateOnLeave}
+      show={show}
+      style={userStyle}
+      {...rest}
+    >
+      {expandable ? (
+        <TouchableWithoutFeedback
+          accessibilityComponentType="button"
+          accessibilityLabel="expand"
+          onPress={onToggleExpanded}
+        >
+          {mainContent}
+        </TouchableWithoutFeedback>
+      ) : (
+        mainContent
+      )}
+    </AnimateAndFade>
+  );
+};
+
+BpkBannerAlert.propTypes = {
+  message: PropTypes.string.isRequired,
+  type: PropTypes.oneOf(Object.keys(ALERT_TYPES).map(key => ALERT_TYPES[key]))
+    .isRequired,
+  animateOnEnter: PropTypes.bool,
+  animateOnLeave: PropTypes.bool,
+  children: PropTypes.node,
+  dismissable: dismissablePropType,
+  dismissButtonLabel: PropTypes.string,
+  expanded: PropTypes.bool,
+  onDismiss: PropTypes.func,
+  onToggleExpanded: PropTypes.func,
```

Comment:
I would almost expect the signature of this to be `(isExpanded) => { ... }`

Reply:
that would be a breaking change and the iOS counterpart will need to change, that's not in the scope of this PR

Reply:
Ah of course, makes sense